src/main/kotlin/onl/tesseract/srp/controller/item/CustomItemDropListener.kt

```diff
@@ -0,0 +1,51 @@
+package onl.tesseract.srp.controller.item
+
+import onl.tesseract.srp.domain.item.CustomItemStack
+import onl.tesseract.srp.domain.item.CustomMaterial
+import onl.tesseract.srp.domain.item.CustomMaterialBlockSource
+import onl.tesseract.srp.domain.item.CustomMaterialEntitySource
+import onl.tesseract.srp.service.item.CustomItemService
+import org.bukkit.event.EventHandler
+import org.bukkit.event.Listener
+import org.bukkit.event.block.BlockBreakEvent
+import org.bukkit.event.entity.EntityDeathEvent
+import org.springframework.stereotype.Component
+
+@Component
+class CustomItemDropListener(private val customItemService: CustomItemService) : Listener {
+
+    @EventHandler
+    fun onBlockBreak(event: BlockBreakEvent) {
+        val player = event.player
+        val block = event.block
+        val material = block.type
+
+        val customMaterial = CustomMaterial.entries.find {
+            it.dropSource is CustomMaterialBlockSource && (it.dropSource).material == material
+        }
```

Comment:
Tu peux ajouter directement `?: return` pour ne pas avoir à faire le if en dessous.
Du genre
```suggestion
        val customMaterial = CustomMaterial.entries
            .find { it.dropSource is CustomMaterialBlockSource && (it.dropSource).material == material }
            ?: return
```